exec/vector/src/main/java/org/apache/drill/exec/vector/BitVector.java

```diff
@@ -285,35 +285,55 @@ public void transferTo(BitVector target) {
 
   public void splitAndTransferTo(int startIndex, int length, BitVector target) {
     assert startIndex + length <= valueCount;
-    int firstByte = getByteIndex(startIndex);
-    int byteSize = getSizeFromCount(length);
-    int offset = startIndex % 8;
-    if (offset == 0) {
+    int firstByteIndex = getByteIndex(startIndex);//byte offset of the first src byte
+    int numBytesHoldingSourceBits = getSizeFromCount(length); //src bytes to read (including start/end bytes that might not be fully copied)
+    int firstBitOffset = startIndex % 8; //Offset of first src bit within the first src byte
+    if (firstBitOffset == 0) {
       target.clear();
       // slice
       if (target.data != null) {
         target.data.release();
       }
-      target.data = data.slice(firstByte, byteSize);
+      target.data = data.slice(firstByteIndex, numBytesHoldingSourceBits);
```

Comment:
I think this is true if `(firstBitOffset == 0 && length%8 == 0)` `if length%8 !=0` then we are keeping reference of more bits then requested. But as memory for underlying Drillbuf is always allocated at byte level then I guess it doesn't make any difference even if we copy extra bits.

However in target vector we are setting the `valueCount` same as `length` but it's not being used while getting any value from Vector (or in `get` apis). I think we should put that check as bounds checking. The internal Drillbuf bounds checking will not catch it since length of buffer is larger than actual data which will allow accessing a value beyond `ValueCount`.

Reply:
Good point. I have added a check.